pilot/test/mock/config.go

```diff
@@ -516,12 +519,21 @@ func CheckCacheEvents(store model.ConfigStore, cache model.ConfigStoreCache, nam
 	CheckMapInvariant(store, t, namespace, n)
 
 	log.Infof("Waiting till all events are received")
-	util.Eventually(func() bool {
-		lock.Lock()
-		defer lock.Unlock()
-		return added == n && deleted == n
-
-	}, t)
+	timeout := time.After(60 * time.Second)
+	for {
+		select {
+		case <-timeout:
+			t.Fatalf("timeout waiting to receive expected events. actual added %d deleted %d. expected %d",
+				added, deleted, n)
+		default:
+			lock.Lock()
+			if added == n && deleted == n {
+				return
+			}
+			lock.Unlock()
```

Comment:
hmm ... you also need to unlock when you return above.

Will `defer` statements be executed after leaving a block or a function.  In other words, can we do this?

```go
{
  lock.Lock()
  defer lock.Unlock()
  if added == n && deleted == n {
    return
  }
}
cond.Wait()
```

Reply:
PTAL
I think we can put defer there. but IDE complains about using defer in for loop so i use explicit unlock.